### DISCUSSION TITLE
Login form doesn't redirect anywhere

### DISCUSSION THREAD

**Discussion by milek8651** [symfony/symfony#46575] [Q&A]
Hi, I made a registration form with make:registration-form command and it works - it puts the new user in the database, but I can't get the login form to work. I made it with make:auth and when I type the correct credentials it doesn't redirect me to the TODO page nor my custom route. I'm using symfony 6.1

**Comment by alessandro-podo** [comment]
can you post your code (authenticator) ?

**Reply by milek8651** [reply]
`<?php

namespace App\Security;

use Symfony\Component\HttpFoundation\RedirectResponse;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
use Symfony\Component\Security\Core\Security;
use Symfony\Component\Security\Http\Authenticator\AbstractLoginFormAuthenticator;
use Symfony\Component\Security\Http\Authenticator\Passport\Badge\CsrfTokenBadge;
use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
use Symfony\Component\Security\Http\Authenticator\Passport\Credentials\PasswordCredentials;
use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
use Symfony\Component\Security\Http\Util\TargetPathTrait;

class LoginFormAuthenticator extends AbstractLoginFormAuthenticator
{
    use TargetPathTrait;

    public const LOGIN_ROUTE = 'app_login';

    private UrlGeneratorInterface $urlGenerator;

    public function __construct(UrlGeneratorInterface $urlGenerator)
    {
        $this->urlGenerator = $urlGenerator;
    }

    public function authenticate(Request $request): Passport
    {
        $email = $request->request->get('email', '');

        $request->getSession()->set(Security::LAST_USERNAME, $email);

        return new Passport(
            new UserBadge($email),
            new PasswordCredentials($request->request->get('password', '')),
            [
                new CsrfTokenBadge('authenticate', $request->request->get('_csrf_token')),
            ]
        );
    }

    public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
    {
        if ($targetPath = $this->getTargetPath($request->getSession(), $firewallName)) {
            return new RedirectResponse($targetPath);
        }

        // For example:
        //return new RedirectResponse($this->urlGenerator->generate('some_route'));
        return new RedirectResponse($this->urlGenerator->generate('app_posts'));
    }

    protected function getLoginUrl(Request $request): string
    {
        return $this->urlGenerator->generate(self::LOGIN_ROUTE);
    }
}`

it's the make:auth default

**Reply by milek8651** [reply]
and also
 `<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Security\Http\Authentication\AuthenticationUtils;

class SecurityController extends AbstractController
{
    #[Route(path: '/login', name: 'app_login')]
    public function login(AuthenticationUtils $authenticationUtils): Response
    {
        // if ($this->getUser()) {
        //     return $this->redirectToRoute('target_path');
        // }

        // get the login error if there is one
        $error = $authenticationUtils->getLastAuthenticationError();
        // last username entered by the user
        $lastUsername = $authenticationUtils->getLastUsername();

        return $this->render('security/login.html.twig', ['last_username' => $lastUsername, 'error' => $error]);
    }

    #[Route(path: '/logout', name: 'app_logout')]
    public function logout(): void
    {
        throw new \LogicException('This method can be blank - it will be intercepted by the logout key on your firewall.');
    }
}
` 
it's SecurityController.php

**Reply by alessandro-podo** [reply]
i dont see a error. 

can you try to dump/breakpoint to find which works and where the error is

**Reply by milek8651** [reply]
how? i dont know where the logs are and it doesn't return any login error
can it be related to the sqlite database?

**Reply by alessandro-podo** [reply]
try in onAuthenticationSuccess 
dump($request) before the if statement

then try a login and look in the debugbar, if you see your dump 

default the logs are var/log/<enviroment>

**Reply by milek8651** [reply]
from /var/log:
"[...] [debug] checking for authenticator support, referer http://......."
"[...] [debug] checking support on authenticator, referer http://......."
"[...] [debug] authenticator does not support the request, referer http://......."

**Comment by lyrixx** [comment]
Hello, I think you best option if you are stuck, is to [create a reproducer](https://symfony.com/doc/current/contributing/code/reproducer.html#reproducing-complex-bugs)
